src/Main.hs

```diff
@@ -70,6 +70,7 @@ data SwitchMethod
     deriving (Eq, Show, ParseFields)
 
 instance ParseField SwitchMethod where
+  readField = Options.readerError "Internal, fatal error: unexpected use of readField"
```

Comment:
Alternatively, you can derive `Read` for `SwitchMethod` and `Direction`, even if it's not used.  Then you can omit the `readField` implementation and it will default to using the `Read` instance

Reply:
Probably down to preference, but I guess I slightly prefer the explicit error since it better captures the expectation that `readField` invocation was indeed unexpected. If the derived `Read` instances start being used elsewhere by `optparse-generic` at some point in the future, it could introduce an unintended execution path. I'll admit that this might be overly defensive, though...

Reply:
I prefer the explicit `readField` implementation too, I think it's clearer.

Reply:
Originally I had derived `Read` instances to solve this issue but it didn't feel good to me when I was about to submit the PR because it wasn't very clear about what was going on.